### DISCUSSION TITLE
Podman Desktop Preferences

### DISCUSSION THREAD

**Discussion by LtCmdrScott** [podman-desktop/podman-desktop#2821] [Q&A] [answered]
Hello All!

I work on a large corporate environment where the users workstations are locked down pretty tight and as such I have to install the Airgapped version of Podman as users are not able to download from GitHub directly (don't get me started on that...), my question is...where are the prefs file settings stored on macOS?  I have to deliver a "preset" version of Podman that already has our proxy enabled on it, but I cannot find were the files are located at...

TIA

**Comment by benoitf** [comment, marked as answer]
hello @LtCmdrScott 

prefs file is at `~/.local/share/containers/podman-desktop/configuration/settings.json` on macOS

**Reply by LtCmdrScott** [reply]
thank you so much for the fast reply!

**Comment by LtCmdrScott** [comment]
I do have a few more questions in regards to this "non-admin" setup that I have to create.  If I want to preinstall the docker compatibility mode" what installer do I need to run ahead of time (or during install) so that the user isn't hit with that installation option when they launch Podman?  Are there any other installers that I should be running during installation time so that the user don't get hit with an additional installer after they launch?

**Comment by benoitf** [comment]
@LtCmdrScott during the installation of podman (not podman-desktop) on macOS the podman-mac-helper is automatically launched (this is the one that provides the 'docker compatibility/ create /var/run/docker.sock file)

https://github.com/containers/podman/releases/download/v4.5.1/podman-installer-macos-arm64.pkg

https://github.com/containers/podman/blob/main/contrib/pkginstaller/scripts/postinstall#L11

**Reply by LtCmdrScott** [reply]
thank you SOO much for your assistance in this!  I really really appreciate it...

**Comment by LtCmdrScott** [comment]
When I try to do this command from my script: (run with sudo as a post install script)

LoggedInUser=$( echo "show State:/Users/ConsoleUser" | /usr/sbin/scutil | /usr/bin/awk '/Name :/ && ! /loginwindow/ { print $3 }' )
export SUDO_USER="${LoggedInUser}"

echo "Installing Helper"
sudo /opt/podman/bin/podman-mac-helper install

I am getting this error and the installer never happens:

Installing Podman Helper app
Installing Helper
**Error: unexpected root user**

do I need to install this a diff way?

**Comment by benoitf** [comment]
https://github.com/containers/podman/blob/fbf7c27e613665610d17e5fde43490aff3608b36/cmd/podman-mac-helper/main.go#L100

I guess command requires to know for which user it needs to add compatibilty mode

so by running it using `root` user it fails

If you know the 'target' user, probably should use `SUDO_USER` or `USER` or env variable when executing the command (and probably you don't need sudo command if you're already root)
